### ISSUE TITLE
ASI should be triggered for identifier named 'abstract'

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#4611]
`abstract` is only a *FutureReservedWord* in ECMAScript 3. In ES5/6, the following code:

```JavaScript
abstract
class C {
}
```

should be parsed as

```JavaScript
abstract;
class C {
}
```

due to ASI rules.

Similarly, one could consider that the following:

```TypeScript
abstract class C {
    abstract
    s();
}
```

should be parsed as

```TypeScript
abstract class C {
    abstract;
    s();
}
```

**Comment by sandersn** [comment]
async doesn't work inside class declarations either:

    async
    s();

is parsed as

    async s();

**Comment by DanielRosenwasser** [comment]
Pinging @bterlson; @rbuckton would like to know whether the *AsyncMethod* production should hold even in the presence of a `static` modifier:

> *AsyncMethod*
&nbsp;&nbsp;`async` [no *LineTerminator* here] *PropertyName*<sub>[?Yield, ?Await]</sub> `(`*StrictFormalParameters*<sub>[Await]</sub>`)`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`{` *AsyncFunctionBody* `}`

**Comment by DanielRosenwasser** [comment]
@sandersn I think we should make all class modifiers (apart from `static`) have a "[no LineTerminator here]".

**Comment by weswigham** [comment]
@DanielRosenwasser we allow newlines after `get` and `set`:
```ts
class Foo {
  public
  static
  create(): Foo { return new Foo(); }

  protected
  get
  j(): number {
    return 42
  }
}
```

**Comment by mhegazy** [comment]
@sandersn can you add blurb on this in https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes.

**Comment by bterlson** [comment]
I'm not sure if this is still relevant but... yes, the production for async methods should be identical with and without a static modifier (and in both cases should not allow a line break after async). While not technically necessary, since it is necessary for async function decls I thought being consistent is better.

**Comment by DanielRosenwasser** [comment]
> I thought being consistent is better.

:+1: :+1:

**Comment by sandersn** [comment]
The change is now documented on the breaking changes list.
